agent/lib/broker_address_settings.js

```diff
@@ -86,34 +111,35 @@ BrokerAddressSettings.prototype.delete_settings = function (plan) {
 };
 
 BrokerAddressSettings.prototype.updated = function (objects) {
-    var plans = objects.map(extract_address_plan).filter(required_broker_resource);
-    plans.sort(kubernetes_resource_compare);
-    var changes = myutils.changes(this.last, plans, kubernetes_resource_compare, same_broker_resource);
+    var plans = objects.map(extract_flatten_plan).filter(required_broker_resource);
+    plans.sort(addressplan_name_compare);
+    var changes = myutils.changes(this.last, plans, addressplan_name_compare, same_broker_resource);
     this.last = plans;
     if (changes) {
         log.info('address plans: %s', changes.description);
         changes.added.map(this.update_settings.bind(this));
         changes.modified.map(this.update_settings.bind(this));
         changes.removed.map(this.delete_settings.bind(this));
+        if (changes.modified.length) {
+            this.dispatch('addressplans_changed', changes.modified, changes.description);
+        }
     }
 };
 
 BrokerAddressSettings.prototype.get_address_settings_async = function (address, global_max_size_promise) {
     var self = this;
-    if (this.last === undefined) {
-        return this.wait_for_plans().then(function () {
-            return global_max_size_promise.then(function (global_max_size) {
-                var settings = self.generate_address_settings(address.plan, global_max_size);
-                log.info('using settings %j for %s', settings, address.address);
-                return settings;
-            });
-        });
-    } else {
+
+    var max_size_as_bytes = this.config.BROKER_GLOBAL_MAX_SIZE ? myutils.parseToBytes(this.config.BROKER_GLOBAL_MAX_SIZE) : 0;
```

Comment:
Would it be cleaner to read GLOBAL_MAX_SIZE/request max size from the Broker once in `start()`  After all the value is not going to change.  

something like:

```
BrokerAddressSettings.prototype.start = function () {
...
read_global_max_size().then((size) => {
self.global_max_szie = size;
self.watcher.on('updated', self.updated.bind(this));
});
```